src/main/java/org/prebid/server/bidder/richaudience/RichaudienceBidder.java

```diff
@@ -177,24 +218,34 @@ private List<BidderBid> extractBids(BidResponse bidResponse, BidRequest bidReque
             return Collections.emptyList();
         }
 
+        final Map<String, Imp> impMap = bidRequest.getImp().stream()
+                .collect(Collectors.toMap(Imp::getId, Function.identity()));
+
         return bidResponse.getSeatbid().stream()
                 .filter(Objects::nonNull)
-                .map(SeatBid::getBid)
+                .flatMap(seatBid -> seatBid.getBid().stream())
                 .filter(Objects::nonNull)
-                .flatMap(Collection::stream)
-                .map(bid -> BidderBid.of(bid,
-                        resolvedBidType(bid.getImpid(), bidRequest.getImp()), bidResponse.getCur()))
+                //todo: is it fine to return a bid with unknown bid type? (as the PBS Go does)
```

Comment:
@SerhiiNahornyi any comments on that?

Reply:
Yes, it's fine, you can check `ResponseBidValidator.java` to double check this.
But, PBS Go actually returns bid type for correct imps.